libflux/src/flux/semantic/analyze.rs

```diff
@@ -1267,7 +1271,11 @@ mod tests {
                                     value: Expression::Duration(DurationLit {
                                         loc: b.location.clone(),
                                         typ: type_info(),
-                                        value: chrono::Duration::minutes(10),
+                                        value: Duration {
+                                            months: -1,
+                                            nanoseconds: -50,
```

Comment:
I expected these values to be positive as per the issue description https://github.com/influxdata/flux/issues/2290 Meaning that the `months` and `nanoseconds` values are always positive values and the `negative` field determines if they should be treated as negative.

Reply:
Yes, that makes sense. I'll make that change.